src/data/live.ts

```diff
@@ -0,0 +1,130 @@
+import {Observable} from 'rxjs'
+
+import type {ObservableSanityClient, SanityClient} from '../SanityClient'
+import type {Any, LiveEventMessage, LiveEventRestart} from '../types'
+import {_getDataUrl} from './dataMethods'
+
+/**
+ * @alpha this API is experimental and may change or even be removed
+ */
+export class LiveClient {
+  #client: SanityClient | ObservableSanityClient
+  constructor(client: SanityClient | ObservableSanityClient) {
+    this.#client = client
+  }
+
+  events(): Observable<LiveEventMessage | LiveEventRestart> {
+    const path = _getDataUrl(this.#client, 'live/events')
+    const url = new URL(this.#client.getUrl(path, false))
+
+    const listenFor = ['restart', 'message'] as const
+
+    return new Observable((observer) => {
+      let es: InstanceType<typeof EventSource> | undefined
+      let reconnectTimer: NodeJS.Timeout
+      let stopped = false
+      // Unsubscribe differs from stopped in that we will never reopen.
+      // Once it is`true`, it will never be `false` again.
+      let unsubscribed = false
+
+      open()
+
+      // EventSource will emit a regular event if it fails to connect, however the API will emit an `error` MessageEvent if the server goes down
+      // So we need to handle both cases
+      function onError(evt: MessageEvent | Event) {
+        if (stopped) {
+          return
+        }
+
+        // If the event has a `data` property, then it`s a MessageEvent emitted by the API and we should forward the error and close the connection
+        if ('data' in evt) {
+          try {
+            const event = parseEvent(evt)
+            observer.error(new Error(event.message, {cause: event}))
+          } catch (err) {
+            observer.error(err)
+          }
```

Comment:
nit: is this catch necessary?  `parseEvent` catches errors as well and returns the error, in which case this catch wouldn't be triggered and `event` on line 43 may not match the expected structure.

Reply:
Thanks, fixing!